component/grant-type/src/main/java/org/wso2/carbon/identity/oauth2/grant/token/exchange/TokenExchangeGrantHandler.java

```diff
@@ -0,0 +1,326 @@
+                log.debug("IAT validity period is set to: " + validityPeriod + " minutes for JWT grant.");
+            }
+        }
+    }
+
+    /**
+     * Validate the Token Exchange Grant.
+     * Checks whether the token request satisfies the requirements to exchange the token.
+     *
+     * @param tokReqMsgCtx OAuthTokenReqMessageContext
+     * @return true or false if the grant_type is valid or not.
+     * @throws IdentityOAuth2Exception Error when validating the Token Exchange Grant
+     */
+    @Override
+    public boolean validateGrant(OAuthTokenReqMessageContext tokReqMsgCtx) throws IdentityOAuth2Exception {
+
+        SignedJWT signedJWT;
+        IdentityProvider identityProvider;
+        String tokenEndPointAlias;
+        JWTClaimsSet claimsSet = null;
+        String requested_audience = null;
+        boolean audienceFound;
+
+        RequestParameter[] params = tokReqMsgCtx.getOauth2AccessTokenReqDTO().getRequestParameters();
+        Map<String, String[]> requestParams = Arrays.stream(params).collect(Collectors.toMap(RequestParameter::getKey,
+                RequestParameter::getValue));
+        if (requestParams.get(TokenExchangeConstants.REQUESTED_TOKEN_TYPE) != null) {
+            requested_token_type = requestParams.get(TokenExchangeConstants.REQUESTED_TOKEN_TYPE)[0];
+        }
+        if (requestParams.get(TokenExchangeConstants.AUDIENCE) != null) {
+            requested_audience = requestParams.get(TokenExchangeConstants.AUDIENCE)[0];
+        }
+        String tenantDomain = tokReqMsgCtx.getOauth2AccessTokenReqDTO().getTenantDomain();
+        if (StringUtils.isEmpty(tenantDomain)) {
+            tenantDomain = MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
+        }
+        if (TokenExchangeConstants.JWT_TOKEN_TYPE.equals(requested_token_type)) {
+            signedJWT = TokenExchangeUtils.getSignedJWT(requestParams.get(TokenExchangeConstants.SUBJECT_TOKEN)[0]);
+            if (signedJWT == null) {
+                handleException(OAuth2ErrorCodes.INVALID_REQUEST, "No Valid subject token was found for "
+                        + TokenExchangeConstants.TOKEN_EXCHANGE_GRANT_TYPE);
+            } else {
+                claimsSet = TokenExchangeUtils.getClaimSet(signedJWT);
+            }
+
+            if (claimsSet == null) {
+                handleException(OAuth2ErrorCodes.INVALID_REQUEST, "Claim values are empty in the given JSON Web Token");
+            }
+
+            String jwtIssuer = claimsSet.getIssuer();
+            String subject = resolveSubject(claimsSet);
+            List<String> audiences = claimsSet.getAudience();
+            Map<String, Object> customClaims = new HashMap<>(claimsSet.getClaims());
+
+            tokReqMsgCtx.addProperty(TokenExchangeConstants.EXPIRY_TIME, claimsSet.getExpirationTime());
+
+            validateRequiredClaims(claimsSet, subject);
+            identityProvider = TokenExchangeUtils.getIdPByIssuer(jwtIssuer, tenantDomain);
```

Comment:
Shall we static import the TokenExchangeUtils class since we are frequently using the util methods from this class and it would improve readability